src/wristpy/core/orchestrator.py

```diff
@@ -0,0 +1,259 @@
+                True the dataframe will contain enmo_epoch1, anglez_epoch1,
+                nonwear_epoch1, physical_activity_levels and sleep_windows_epoch1. If
+                False the the dataframe will be in "raw" time and contain the enmo and
+                anglez data.
+
+        Returns:
+            The polars dataframe to be saved into csv or parquet format.
+
+        Raises:
+            ValueError if the attribute for the time column is None.
+        """
+        data_source = self.enmo_epoch1 if use_epoch1_time else self.enmo
+
+        if data_source is None:
+            raise ValueError(f"{data_source} is None, can't construct time column.")
+
+        time_data = data_source.time
+
+        results_dataframe = pl.DataFrame({"time": time_data})
+
+        for field_name, field_value in self:
+            if field_value is not None:
+                if (
+                    use_epoch1_time
+                    and field_name
+                    not in ["enmo", "anglez", "sleep_windows", "nonwear_array"]
+                ) or (not use_epoch1_time and field_name in ["enmo", "anglez"]):
+                    results_dataframe = results_dataframe.with_columns(
+                        pl.Series(field_name, field_value.measurements)
+                    )
+
+        return results_dataframe
+
+
+def format_sleep_data(
+    sleep_windows: List[analytics.SleepWindow], reference_measure: models.Measurement
+) -> np.ndarray:
+    """Formats sleep windows into an array for saving.
+
+    Args:
+        sleep_windows: The list of time stamp pairs indicating periods of sleep.
+        reference_measure: The measure from which the temporal resolution will be taken.
+            Any epoch1 measure can be used, but enmo is used for consistency.
+
+    Returns:
+        1-D binary np.ndarray, with 1 indicating sleep. Will be of the same length as
+            the timestamps in the epoch1_measure.
+    """
+    sleep_array = np.zeros(len(reference_measure.time))
+
+    for window in sleep_windows:
+        sleep_mask = (reference_measure.time >= window.onset) & (
+            reference_measure.time <= window.wakeup
+        )
+        sleep_array[sleep_mask] = 1
+
+    return sleep_array
+
+
+def format_nonwear_data(
+    nonwear_data: models.Measurement, reference_measure: models.Measurement
+) -> np.ndarray:
+    """Format nonwear data into an array for saving.
+
+    Args:
+        nonwear_data: The nonwear measurement.
+        reference_measure: The measure from which the temporal resolution will be taken.
+            Any epoch1 measure can be used, but enmo is used for consistency.
+
+    Returns:
+        1-D binary np.ndarray, with 1 indicating nonwear. Will be of the same length as
+            the timestamps in the reference_measure.
+
+    """
+    nonwear_df = pl.DataFrame(
+        {
+            "nonwear": nonwear_data.measurements.astype(np.int64),
```

Comment:
Docstring claims you're outputting a binary, but you're building an int64. Does polars allow casting as boolean?

Reply:
This was me being silly, when i said binary here i meant in terms of the possible values only being 1s and 0s, not the data type. I'll remove that to prevent confusion.